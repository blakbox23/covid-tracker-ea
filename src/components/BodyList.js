/* eslint react/prop-types: 0 */import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { v4 as uuidv4 } from 'uuid';
import { IconContext } from 'react-icons';
import { GiAfrica } from 'react-icons/gi';
import Nav from './Nav';
import { getCountries } from '../redux/countries/countries';

function BodyList() {
  const countries = useSelector((state) => state.data.countries);
  console.log('countries');
  console.log(countries);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getCountries());
  }, []);

  return (
    <>
      <Nav txt="Home Navbar" />
      <header className="header flex">
        <div className="header-img">
          <IconContext.Provider value={{ className: 'africa-img' }}>
            <div>
              <GiAfrica />
            </div>
          </IconContext.Provider>
        </div>
        <div className="header-text">
          <div><b>AFRICA</b></div>
          <div>num</div>
        </div>
      </header>
      <p className="title">COVID NUMBERS IN AFRICA</p>
      <ul>
        {countries.map((country) => (
          <li key={uuidv4()}>
            <Link to={`/${country.country}`}>
              {country.country}
            </Link>
            <div>{country.cases}</div>
          </li>
        ))}
      </ul>

      {/* <ul>
        {data.countries.map((country) => (
          <li key={region.id}>
            <Link to={`/${region.id}`}>
              {region.name}
            </Link>
          </li>
        ))}
      </ul>   */}
    </>
  );
}

export default BodyList;
