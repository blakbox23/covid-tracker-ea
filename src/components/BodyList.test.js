import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import BodyList from './BodyList';
import store from '../redux/configureStore';

describe('Home snapshot', () => {
  test('renders correctly', () => {
    const tree = renderer
      .create(
        <Router>

          <Provider store={store}>
            <BodyList />
          </Provider>
        </Router>
        ,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
