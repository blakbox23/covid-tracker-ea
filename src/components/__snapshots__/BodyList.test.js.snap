// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshot renders correctly 1`] = `
Array [
  <div
    className="navbar flex"
  >
    <div />
    <div
      className="logo-txt"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Covid in Africa
      </a>
    </div>
    <a
      className="logo"
      href="/info"
      onClick={[Function]}
    >
      <div>
        <svg
          className="src-logo"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm0-2a8 8 0 1 0 0-16 8 8 0 0 0 0 16zM11 7h2v2h-2V7zm0 4h2v6h-2v-6z"
            />
          </g>
        </svg>
      </div>
    </a>
  </div>,
  <header
    className="header flex"
  >
    <div
      className="header-img"
    >
      <div>
        <svg
          className="africa-img"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M201.56 19.495l-87.79 9.131-73.745 94.814v52.676l56.186 61.805 64.615-13.344 49.164 9.832-10.535 37.926 33.711 61.103-16.855 42.842 39.79 116.225 53.62-8.768 49.164-55.484 4.213-38.629 31.605-23.879-6.322-69.531 83.594-106.994-51.989 7.263-79.363-138.359-125.016-8.428-14.046-30.2zm252.346 319.8l-14.402 20.86-13.408.496c-11.849 24.321-12.598 38.019-13.907 66.547l17.383 4.471 21.852-52.147 2.482-40.226z"
          />
        </svg>
      </div>
    </div>
    <div
      className="header-text"
    >
      <div>
        <b>
          AFRICA
        </b>
      </div>
      <div>
        
      </div>
    </div>
  </header>,
  <p
    className="title"
  >
    TOTAL COVID CASES IN AFRICA
  </p>,
  <div
    className="body-list flex"
  />,
]
`;
